### PR TITLE
add response logic for node status updating

### LINKED ISSUE #257: CloudHub didn't response sync message from edge
**What happened**:

At edge side, edge core logs warning about timeout to receive response.

image

**What you expected to happen**:

No warnings.

**How to reproduce it (as minimally and precisely as possible)**:

common (default) configuration would stable reproduce.

**Anything else we need to know?**:

This is a sync message call, and adds a channel to beehive, waiting for response from cloud hub, but cloud hub not seemed have an response logic for node status updating.

https://github.com/kubeedge/kubeedge/blob/6878d86b9ec33ca716933575b93ef70663504ba5/edge/pkg/edgehub/controller.go#L213-L225

**Environment**:
- KubeEdge version: v0.2
- Hardware configuration: irrelevant
- OS (e.g. from /etc/os-release): irrelevant
- Kernel (e.g. `uname -a`): irrelevant
- Others: irrelevant

### PR DESCRIPTION
**Pull request by shouhong**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Add response logic for node status updating

**Which issue(s) this PR fixes**:
Fixes #257

**Special notes for your reviewer**:

### CODE REVIEW

File: cloud/pkg/controller/controller/upstream.go [hunk 1]
```diff
@@ -331,6 +331,24 @@ func (uc *UpstreamController) updateNodeStatus(stop chan struct{}) {
...
+				nodeID, err := messagelayer.GetNodeID(msg)
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure, get node id failed with error: %s", msg.GetID(), err)
+				}
+				resource, err := messagelayer.BuildResource(nodeID, namespace, model.ResourceTypeNode, name)
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure, build message resource failed with error: %s", msg.GetID(), err)
+				}
+				resMsg.BuildRouter(constants.EdgeControllerModuleName, constants.GroupResource, resource, model.ResponseOperation)
+				err = uc.messageLayer.Response(*resMsg)
```
Comment (m1093782566) [inline cloud/pkg/controller/controller/upstream.go]: condense it to:

if err := uc.messageLayer.Response(*resMsg); err != nil {
  ...
}

to save one line :)

**Comment by m1093782566** [comment]
cc @sids-b @tedli

File: cloud/pkg/controller/controller/upstream.go [hunk 2]
```diff
@@ -331,6 +331,24 @@ func (uc *UpstreamController) updateNodeStatus(stop chan struct{}) {
 					log.LOGGER.Warnf("message: %s process failure, update node failed with error: %s, namespace: %s, name: %s", msg.GetID(), err, getNode.Namespace, getNode.Name)
 					continue
 				}
+
+				resMsg := model.NewMessage(msg.GetID())
+				resMsg.Content = "update node status successfully"
```
Comment (sids-b) [inline cloud/pkg/controller/controller/upstream.go]: nit: in other places in KubeEdge, we send "OK" as response content. Lets keep it consistent

File: cloud/pkg/controller/controller/upstream.go [hunk 3]
```diff
@@ -331,6 +331,24 @@ func (uc *UpstreamController) updateNodeStatus(stop chan struct{}) {
...
+				nodeID, err := messagelayer.GetNodeID(msg)
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure, get node id failed with error: %s", msg.GetID(), err)
+				}
+				resource, err := messagelayer.BuildResource(nodeID, namespace, model.ResourceTypeNode, name)
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure, build message resource failed with error: %s", msg.GetID(), err)
```
Comment (sids-b) [inline cloud/pkg/controller/controller/upstream.go]: add a continue after this line ?

File: cloud/pkg/controller/controller/upstream.go [hunk 4]
```diff
@@ -331,6 +331,24 @@ func (uc *UpstreamController) updateNodeStatus(stop chan struct{}) {
 					log.LOGGER.Warnf("message: %s process failure, update node failed with error: %s, namespace: %s, name: %s", msg.GetID(), err, getNode.Namespace, getNode.Name)
 					continue
 				}
+
+				resMsg := model.NewMessage(msg.GetID())
+				resMsg.Content = "update node status successfully"
+				nodeID, err := messagelayer.GetNodeID(msg)
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure, get node id failed with error: %s", msg.GetID(), err)
```
[the remark this answers is not in the document]
Comment (shouhong) [reply cloud/pkg/controller/controller/upstream.go]: I checked the existing log messages in this file, all of them are started with lower-case letter, such as "stop updateNodeStatus".

[the remark this answers is not in the document]
Comment (m1093782566) [reply cloud/pkg/controller/controller/upstream.go]: You can change them if you want.

**Comment by shouhong** [comment]
@m1093782566 @sids-b I have updated the code according to your comments. Can you please help review again?

**Pull request merged**